# Tâche 4 : Analyse du sytème cristallin 

In [2]:
!pip install mp-api --upgrade
import numpy as np
from mp_api.client import MPRester
#from pymatgen.ext.matproj import MPRester
from pymatgen.core.operations import SymmOp
from pymatgen.symmetry.analyzer import SpacegroupAnalyzer
from pymatgen.electronic_structure.plotter import BSPlotter
from pymatgen.phonon.plotter import PhononBSPlotter
from jupyter_jsmol.pymatgen import quick_view
from lmapr1492 import plot_brillouin_zone, get_plot_bs, get_plot_dos, get_plot_bs_and_dos, get_branch_wavevectors
from plotly.subplots import make_subplots
import plotly.graph_objects as go

Defaulting to user installation because normal site-packages is not writeable
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 97.1/97.1 kB 1.9 MB/s eta 0:00:00ta 0:00:01
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 61.2/61.2 kB 1.1 MB/s eta 0:00:00ta 0:00:01
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 80.1/80.1 kB 2.2 MB/s eta 0:00:00
  Attempting uninstall: mp-api
    Found existing installation: mp-api 0.40.2
    Uninstalling mp-api-0.40.2:
      Successfully uninstalled mp-api-0.40.2


In [3]:
mp_key = "MgnCbwKILRoKeWCsa2tbfn04aBTHkB78"
mp_id = "mp-1747"

In [4]:
with MPRester(mp_key) as m:
    print("Téléchargement du fichier CIF de K2Te..")
    prim_struc = m.get_structure_by_material_id(mp_id)
    print("Téléchargement terminé")
conv_struc = SpacegroupAnalyzer(prim_struc).get_conventional_standard_structure()

Téléchargement du fichier CIF de K2Te..


Retrieving MaterialsDoc documents:   0%|          | 0/1 [00:00<?, ?it/s]

Téléchargement terminé


In [5]:
a=prim_struc.lattice.a
alpha=prim_struc.lattice.alpha
print("Vecteurs de base du réseau direct :")
print(SpacegroupAnalyzer(prim_struc).find_primitive().lattice.rhombohedral(a,alpha).matrix)
print("\n")

Vecteurs de base du réseau direct :
[[5.01066174 0.         2.89290605]
 [1.67022033 4.72409728 2.89290605]
 [0.         0.         5.78581338]]




In [8]:

a=conv_struc.lattice.a
print("Vecteurs de base du réseau réciproque :")
print(SpacegroupAnalyzer(prim_struc).find_primitive().lattice.cubic(a).matrix)
print("\n")

Vecteurs de base du réseau réciproque :
[[8.18237593 0.         0.        ]
 [0.         8.18237593 0.        ]
 [0.         0.         8.18237593]]




In [7]:
spa = SpacegroupAnalyzer(conv_struc,0.1)
#type de maille
maille = spa.get_lattice_type()
#système cristallin
sys = spa.get_crystal_system()
#groupe ponctuel
ponc = spa.get_point_group_symbol()

print("Type de maille de K2Te :", maille)
print("Système cristallin :",sys)
print("Groupe ponctuel :" ,ponc)

Type de maille de K2Te : cubic
Système cristallin : cubic
Groupe ponctuel : m-3m


## Analyse des résulats 

Le $K_2Te$ possède une maille primitive possédant un système Rhomboédrique Primitif dont les angles sont tous égaux à 60° et les cotés sont tous égaux à $5.79\mathring{A}$. Pour la suite des notebooks, nous allons utiliser la maille conventionnelle qui est beaucoup plus simple car il s'agit ici d'une strucuture en Cubique Faces Centrées liée à un groupe ponctuelle $m\overline{3}m$. Tous les angles valent donc 90° et les cotés valent tous $8.18\mathring{A}$. Cela vérifie bien la définition du réseau réciproque qui dit que $\vec{a} \cdot \vec{a}^* = 2\pi$ et que $\vec{a}^*\cdot\vec{b} = 0$, il en va de même pour tous les autres vecteurs de base.

Nous pouvons également apprendre du groupe ponctuelle que la maille conventionnelle possède : 

- Des symétrie de plan miroir le long des directions : $[100], [010], [001]$
- Des Roto-inversions d'ordre 3 le long des axes : $[111], [1\overline{1}\overline{1}], [\overline{1}1\overline{1}], [\overline{1}\overline{1}1]$
- Des symétries de plan miroir le long des directions : $[110], [101], [011], [1\overline{1} 0], [10\overline{1}], [01\overline{1}]$